packages/components/psammead-social-embed/src/Canonical/index.jsx

```diff
@@ -34,6 +34,8 @@ export const providers = {
         window.instgrm.Embeds.process();
       }
     },
+    onLibraryLoad: () =>
+      console.error('onRender callback function not implemented for Instagram'),
```

Comment:
You'll only want to log an error when the consumer provides an `onRender` function for YT and Insta.

To do this you could make the `onRender` default prop `null` again and something like

```suggestion
    onLibraryLoad: onRender => {
      if (onRender) {
        console.error(getOnRenderError('Instagram'))
      }
    }
```

Reply:
sorry for all the changes!

Reply:
you'll also need to check to do `if (onRender) {` in the twitter one too

Reply:
Thanks @jroebu14, these are really helpful suggestions 🙂 Just a query, would we not be able to add the `onRender` condition to the `onLibraryLoad` condition in the `useEffect`?
Like
```jsx
if (onRender && hasLoadedLibrary && onLibraryLoad) {
      onLibraryLoad(onRender);
}
```

then

```jsx
onLibraryLoad: () => getOnRenderError('Instagram'),
```

Reply:
yeh that sounds better! sorry for going a little full circle on this one